tap_dynamodb/sync_strategies/log_based.py

```diff
@@ -115,6 +117,34 @@ def sync_shard(shard, seq_number_bookmarks, streams_client, stream_arn, projecti
     singer.write_state(state)
     return rows_synced
 
+def prepare_projection(projection, expression):
+    '''
+    Prepare the projection based on the expression attributes
+    For example:
+    projections = "#c", expressions = {"#c": "Comment"}
+    return = ["Comment"]
+    Example of nested expression :
+    In Catalog - projections = "#n[0].#a", expressions = {"#n": "Name", "#a": "Age"}
+    projection passed in the function = ['#n[0]', '#a']
+    projection returned from the function = ['Name[0]', 'Age']
+    '''
+    expression = json.loads(expression)
+    return_projection = copy.deepcopy(projection)
+    # iterating over the expressions dict
+    for key, value in expression.items():
+        # iterating over the projections list
+        for proj_index, element in enumerate(projection):
+            if element == key:
+                return_projection[proj_index] = value
+            if '[' in element:
+                # split exactly one time by `[`
+                split_portion = element.split('[', 1)
+                # check if the zeroth position is key and replace
+                if split_portion[0] == key:
+                    # replace the value with the key of expression attributes
+                    return_projection[proj_index] = return_projection[proj_index].replace(key, value)
+    return return_projection
```

Comment:
Why do we need to return anything?

Reply:
removed the return, updated the current list